### ISSUE TITLE
Translation of Validation Error Messages

### ISSUE THREAD

**Issue by marbetschar** [schoero/swissqrbill#441]
Is there a way to translate the error messages?

We use SwissQRCode to generate a QR Code based on dynamic formular data - and it would be nice to provide the user with information about which validation failed in their language.

**Comment by schoero** [comment]
There is no built in way to do this.

But the errors are exported and can be caught and re-thrown with a translated message:

```ts
import { SwissQRBill } from "swissqrbill/pdf";
import { ValidationError, ValidationErrors } from "swissqrbill/errors";

const TRANSLATED_ERRORS_DE: Record<ValidationErrors, string> = {
  [ValidationErrors.AMOUNT_LENGTH_IS_INVALID]: "Der Betrag darf maximal 12 Ziffern enthalten",
  //...
};

try {
  const qrBill = new SwissQRBill(/* data */);
  //...
} catch(error) {
  if(error instanceof ValidationError){
    throw new Error(TRANSLATED_ERRORS_DE[error.message]);
  }
}
```

What bothers me with that solution is that the error message currently has to be used for the mapping.  

I think the best solution would be to add the [enum key](https://github.com/schoero/swissqrbill/blob/78b19f13044197520a722e5e70ed9d25b9221840/src/shared/errors.ts#L19) as an `error.code` property to provide a consistent and reliable reference for error handling.

**Comment by marbetschar** [comment]
I agree, having an enum key would be nice!

**Comment by schoero** [comment]
This is now implemented in v4.1.0 and can be used like this:


```ts
import { SwissQRBill } from "swissqrbill/pdf";
import { ValidationError, ValidationErrors } from "swissqrbill/errors";

const TRANSLATED_ERRORS_DE: Record<ValidationError["code"], string> = {
  AMOUNT_LENGTH_IS_INVALID: "Der Betrag darf maximal 12 Ziffern enthalten",
  //...
};

try {
  const qrBill = new SwissQRBill(/* data */);
  //...
} catch(error) {
  if(error instanceof ValidationError){
    throw new Error(TRANSLATED_ERRORS_DE[error.code]));
  }
}
```

**Comment by marbetschar** [comment]
Nice, thank you!!
